src/App.tsx

```diff
@@ -1,42 +1,107 @@
+import { User } from './types/User';
+import { getUsers } from './services/user';
+import { Post } from './types/Post';
+import { getPostsByUserId } from './services/posts';
+import classNames from 'classnames';
+
+export const App: FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPostId, setSelectedPostId] = useState<number | null>(null);
+  const [postError, setPostError] = useState<boolean>(false);
+  const [isLoadingPosts, setIsLoadingPosts] = useState<boolean>(false);
+
+  const selectedPost = useMemo(
+    () => userPosts.find(post => post.id === selectedPostId),
+    [selectedPostId, userPosts],
+  );
+
+  const setSelectUser = (user: User) => {
+    setSelectedUser(user);
+    setUserPosts([]);
+  };
+
+  const setSelectPostId = (id: number | null) => {
+    setSelectedPostId(id);
+  };
+
+  useEffect(() => {
+    if (selectedUser) {
+      setIsLoadingPosts(true);
+      setPostError(false);
+      getPostsByUserId(selectedUser.id)
+        .then(setUserPosts)
+        .catch(() => setPostError(true))
+        .finally(() => setIsLoadingPosts(false));
+    } else {
+      setUserPosts([]);
+    }
+  }, [selectedUser]);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
 
-export const App: React.FC = () => {
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectUser={setSelectUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                <Loader />
+                {isLoadingPosts && <Loader />}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {userPosts.length > 0 && !isLoadingPosts && (
+                  <PostsList
+                    userPosts={userPosts}
+                    setSelectPostId={setSelectPostId}
+                    selectedPostId={selectedPostId}
+                  />
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {!isLoadingPosts &&
+                  userPosts.length === 0 &&
```

Comment:
```suggestion
                  !userPosts.length &&
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**